test_data/generate_test_data.py

```diff
@@ -0,0 +1,195 @@
+"""
+Generates example data using pyarrow by running `python generate_test_data.py`.
+
+You can print the metadata with:
+
+.. code-block:: python
+
+   >>> import json, pprint, pyarrow.parquet as pq
+   >>> pprint.pprint(json.loads(pq.read_schema("example.parquet").metadata[b"geo"]))
+"""
+import json
+import pathlib
+import copy
+
+import pyarrow as pa
+import pyarrow.parquet as pq
+from pyarrow.csv import write_csv
+
+from shapely import from_wkt, to_wkb
+
+
+HERE = pathlib.Path(__file__).parent
+
+
+metadata_template = {
+    "version": "1.1.0",
+    "primary_column": "geometry",
+    "columns": {
+        "geometry": {
+            "encoding": "WKB",
+            "geometry_type": [],
+        },
+    },
+}
+
+
+## Various geometry types with WKB and native (GeoArrow-based) encodings
+
+def write_encoding_files(geometries_wkt, geometries_geoarrow, geometry_type):
+
+    table = pa.table({"col": range(len(geometries_wkt)), "geometry": geometries_wkt})
+    write_csv(table, HERE / f"data-{geometry_type.lower()}.wkt")
+
+    # WKB encoding
+    table = pa.table(
+        {"col": range(len(geometries_wkt)), "geometry": to_wkb(from_wkt(geometries_wkt))}
+    )
+    metadata = copy.deepcopy(metadata_template)
+    metadata["columns"]["geometry"]["geometry_type"] = [geometry_type]
+    table = table.replace_schema_metadata({"geo": json.dumps(metadata)})
+    pq.write_table(table, HERE / f"data-{geometry_type.lower()}-encoding_wkb.parquet")
+
+    # native (geoarrow) encoding
+    table = pa.table(
+        {"col": range(len(geometries_wkt)), "geometry": geometries_geoarrow}
+    )
+    metadata["columns"]["geometry"]["encoding"] = geometry_type.lower()
+    table = table.replace_schema_metadata({"geo": json.dumps(metadata)})
+    pq.write_table(table, HERE / f"data-{geometry_type.lower()}-encoding_native.parquet")
+
+
+# point
+
+geometries_wkt = [
+    "POINT (30 10)",
+    "POINT EMPTY",
+]
+
+point_type = pa.struct(
+    [
+        pa.field("x", pa.float64(), nullable=False),
+        pa.field("y", pa.float64(), nullable=False)
+    ]
+)
+geometries = pa.array([(30, 10), (float("nan"), float("nan"))], type=point_type)
+
+write_encoding_files(
+    geometries_wkt, geometries, geometry_type="Point"
+)
+
+# linestring
+
+geometries_wkt = [
+    "LINESTRING (30 10, 10 30, 40 40)",
+    "LINESTRING EMPTY",
+]
+
+linestring_type = pa.list_(pa.field("vertices", point_type, nullable=False))
+geometries = pa.array(
+    [[(30, 10), (10, 30), (40, 40)], []], type=linestring_type)
```

Comment:
Maybe it would be less error-prone to construct these geometries from the WKT? Then we don't have to manually check that the WKT is equivalent to here. 

_Or_ maybe it would be better to use GeoJSON and pyarrow's JSON reader to infer geometry columns? Though then you'd need to convert from interleaved to separated coords, which is a bit involved for this test data here 😕

Reply:
Yeah, I actually tried to automate writing this hard-coded data a bit, by parsing the WKT with shapely and converting to GeoJSON. The main problem is that then I need to convert the inner lists to tuples (as json is only lists), to have the construction as a separated struct type work with pyarrow (a tuple element can be converted to a struct, but a list element not). And that gives some convoluted logic ..